File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/entity/ClickEntity.java
/*
******************************* Copyright (c)*********************************\
**
**                 (c) Copyright 2015, Allen, china, shanghai
**                          All Rights Reserved
**
**                          
**                         
**-----------------------------------版本信息------------------------------------
** 版    本: V0.1
**
**------------------------------------------------------------------------------
********************************End of Head************************************\
*/
package com.chad.baserecyclerviewadapterhelper.entity;

/**
 * 文 件 名: ClickEntity
 * 创 建 人: Allen
 * 创建日期: 16/11/1 22:16
 * 邮   箱: AllenCoder@126.com
 * 修改时间：
 * 修改备注：
 */
public class ClickEntity {
    public static final int CLICK_ITEM_VIEW = 1;
    public static final int CLICK_ITEM_CHILD_VIEW = 2;
    public static final int LONG_CLICK_ITEM_VIEW = 3;
    public static final int LONG_CLICK_ITEM_CHILD_VIEW = 4;
    private final int type;

    public ClickEntity(final int type) {
        this.type = type;
    }

    public int getItemType() {
        return type;
    }
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/entity/ClickEntity.kt
package com.allen.kotlinapp.entity

import com.chad.library.adapter.base.entity.MultiItemEntity

/**
 * 文 件 名: ClickEntity
 * 创 建 人: Allen
 * 创建日期: 2017/6/13 15:08
 * 修改时间：
 * 修改备注：
 */
class ClickEntity(var Type: Int) : MultiItemEntity {

    override fun getItemType(): Int {
        return Type
    }

    companion object {
        val CLICK_ITEM_VIEW = 1
        val CLICK_ITEM_CHILD_VIEW = 2
        val LONG_CLICK_ITEM_VIEW = 3
        val LONG_CLICK_ITEM_CHILD_VIEW = 4
        val NEST_CLICK_ITEM_CHILD_VIEW = 5
    }
}
